src/shogun/io/openml/OpenMLData.cpp

```diff
@@ -0,0 +1,271 @@
+	for (const auto type : feat_type_copy)
+	{
+		if (type == ARFFDeserializer::Attribute::STRING)
+			SG_SERROR("Currently cannot process string features!\n")
+	}
+	//	auto result = std::make_shared<CDenseFeatures>();
+	std::shared_ptr<CDenseFeatures<float64_t>> result;
+	bool first = true;
+	for (int i = 0; i < m_feature_types.size(); ++i)
+	{
+		if (i != col_idx && first)
+		{
+			result.reset(m_cached_features->get_feature_obj(i)
+			                 ->as<CDenseFeatures<float64_t>>());
+			first = false;
+		}
+		if (i != col_idx)
+			result.reset(
+			    result
+			        ->create_merged_copy(m_cached_features->get_feature_obj(i))
+			        ->as<CDenseFeatures<float64_t>>());
+	}
+
+	// need to copy data as result is only in the stack and the data
+	// will be gone at the end of the function
+	auto* copy_feat = SG_MALLOC(
+	    float64_t,
+	    m_feature_types.size() * m_cached_features->get_num_vectors());
+	memcpy(
+	    copy_feat, result->get_feature_matrix().data(),
+	    m_feature_types.size() * m_cached_features->get_num_vectors());
+
+	result = std::make_shared<CDenseFeatures<float64_t>>(
+	    copy_feat, m_feature_types.size(),
+	    m_cached_features->get_num_vectors());
+
+	return result;
+}
+
+std::shared_ptr<CLabels> OpenMLData::get_labels()
+{
+	REQUIRE(
+	    !m_default_target_attribute.empty(),
+	    "A default target attribute is required if no label is given!\n")
+	return get_labels(m_default_target_attribute);
+}
+
+std::shared_ptr<CLabels> OpenMLData::get_labels(const std::string& label_name)
+{
+	if (m_cached_labels && label_name == m_cached_label_name)
+		return m_cached_labels;
+
+	if (!m_cached_features)
+		get_data();
+
+	auto find_label =
+	    std::find(m_feature_names.begin(), m_feature_names.end(), label_name);
+	if (find_label == m_feature_names.end())
+		SG_SERROR(
+		    "Requested label \"%s\" not in the dataset!\n", label_name.c_str())
+	auto col_idx = std::distance(m_feature_names.begin(), find_label);
+
+	auto target_label_as_feat =
+	    std::shared_ptr<CFeatures>(m_cached_features->get_feature_obj(col_idx));
+
+	switch (m_feature_types[col_idx])
+	{
+	// real features
+	case ARFFDeserializer::Attribute::REAL:
+	case ARFFDeserializer::Attribute::NUMERIC:
+	case ARFFDeserializer::Attribute::INTEGER:
+	case ARFFDeserializer::Attribute::DATE:
+	{
+		auto casted_feat = std::dynamic_pointer_cast<CDenseFeatures<float64_t>>(
+		    target_label_as_feat);
+		auto labels_vec = casted_feat->get_feature_matrix().get_row_vector(0);
+		auto labels = std::make_shared<CRegressionLabels>(labels_vec);
+		m_cached_labels = labels;
+		m_cached_label_name = label_name;
+		return m_cached_labels;
+	}
+	break;
+		// nominal features
+	case ARFFDeserializer::Attribute::NOMINAL:
+	{
+		auto casted_feat = std::dynamic_pointer_cast<CDenseFeatures<float64_t>>(
+		    target_label_as_feat);
+		auto labels_vec = casted_feat->get_feature_matrix().get_row_vector(0);
+		for (auto& val : labels_vec)
+		{
+			if (val == 0)
```

Comment:
maybe it should be an option how those are encoded?

Reply:
yea, this isn't great, it was just to make things work... It would be good to implement more label/feature conversions inside the machine classes